dashboard/app/controllers/pd/workshop_daily_survey_controller.rb

```diff
@@ -21,7 +21,7 @@ def new_general
       day = params[:day].to_i
       return render_404 if day < 0
 
-      workshop = Workshop.
+      workshop = params[:enrollmentCode].present? ? Pd::Enrollment.find_by!(code: params[:enrollmentCode]).workshop : Workshop.
```

Comment:
This is kind of difficult to decipher with the ternary and multi-line query. How about something more like:

```ruby
workshop = if params[:enrollmentCode].present?
  Enrollment.find_by!(code: params[:enrollmentCode]).workshop
else
  Workshop.
    where...
end
```